rest/tests/test_course.py

```diff
@@ -53,6 +61,169 @@ def test_list_courses(self):
+        Note: learning resources types are created during the import of the
+        course but they are related to the repository.
+        """
+        # trying to delete non existing courses
+        self.delete_course(
+            repo_slug=self.repo.slug,
+            course_id='foo',
+            expected_status=404
+        )
+        self.delete_course(
+            repo_slug=self.repo.slug,
+            course_id=1234567,
+            expected_status=404
+        )
+
+        # environment before importing a course
+        self.count_resources(
+            repositories=1,
+            courses=1,
+            learning_resources=1,
+            learning_resource_types=1
+        )
+
+        # Import the course tarball
+        self.import_course_tarball(self.repo)
+        # create a vocabulary and term
+        vocabulary = self.create_vocabulary(self.repo.slug)
+        self.create_term(self.repo.slug, vocabulary['slug'])
+
+        # environment before deleting the course
+        self.count_resources(
+            repositories=1,
+            courses=2,
+            learning_resources=19,
+            learning_resource_types=8,
+            static_assetts=5,
+            terms=1,
+            vocabularies=1
+        )
+
+        # make sure of the course that is going to be deleted
+        courses = Course.objects.all()
+        self.assertEqual(courses[0].id, self.course.id)
+        self.assertNotEqual(courses[1].id, self.course.id)
+
+        # try to get the right course but non existing repo
+        self.delete_course(
+            repo_slug='nonsense',
+            course_id=courses[1].id,
+            expected_status=404
+        )
+
+        # delete the course
+        self.delete_course(self.repo.slug, courses[1].id)
+
+        # environment after deleting the course
+        self.count_resources(
+            repositories=1,
+            courses=1,
+            learning_resources=1,
+            learning_resource_types=8,
+            terms=1,
+            vocabularies=1
+        )
+
+        # the course remaining is the one not deleted
+        courses = Course.objects.all()
+        self.assertEqual(courses[0].id, self.course.id)
+
+        # and the learning resource remaining is associated with the remaining
+        # course
+        learning_resources = LearningResource.objects.all()
+        self.assertEqual(
+            learning_resources[0].course.id,
+            self.course.id
+        )
+
+        # try to delete the remaining course but with the wrong repo
+        # create another repo
+        another_repo_dict = {
+            'name': 'another_repo_name',
+            'description': 'description',
+        }
+        repo_res = self.create_repository(another_repo_dict)
+
+        # try to get the right course but on wrong repo
+        self.delete_course(
+            repo_slug=repo_res['slug'],
+            course_id=self.course.id,
+            expected_status=404
+        )
+
```

Comment:
Do you have a test to check that, given a course c1 in repo r1 and course c2 in repo c2, that /repository/r1/course/c2 and /repository/r2/course/c1 gets a 404?